activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb

```diff
@@ -37,7 +31,8 @@ def postgresql_connection(config)
       conn_params[:dbname] = conn_params.delete(:database) if conn_params[:database]
 
       # Forward only valid config params to PGconn.connect.
-      conn_params.keep_if { |k, _| VALID_CONN_PARAMS.include?(k) }
+      valid_conn_params = PGconn.conndefaults_hash.keys + [:requiressl]
+      conn_params.keep_if { |k, _| valid_conn_params.include?(k) }
```

Comment:
better name would be `valid_conn_param_keys`.

also, if active support is available, you can do `conn_params.slice!(valid_conn_param_keys)`